### PR TITLE
Improvements on maven

### PR DESCRIPTION
**Pull request by fcrespo82**
Support for repositories defined in pom.xml (not only settings and central)
Support for local repository (case when a package is not in a repository but only in .m2/repository)
Better tagged version
Show when a package is not found
Use maven binary from path to get effective settings based

Let me know if you have any concerns about this.

### CODE REVIEW

File: src/providers/maven/mavenAPI.js [hunk 1]
```diff
@@ -1,71 +1,92 @@
...
-        repositories.push(MAVEN_CENTRAL)
-      } else {
-        let xml = new xmldoc.XmlDocument(data.toString());
-        let repositoriesXml = xml.descendantWithPath("profiles.profile.repositories").childrenNamed("repository")
+export function loadMavenRepositories() {
+  if (repositories.length > 0) {
+    return Promise.resolve(repositories)
+  }
+  return Promise.all([
+    exec('/bin/bash mvn help:effective-settings | grep -Ev "(\\\[INFO\\\]|\\\[DEBUG\\\])" | tail -n+2').then(response => {
```
Comment (pflannery) [inline src/providers/maven/mavenAPI.js]: This wouldn't work on windows without bash installed ?

**Comment by pflannery** [comment]
@fcrespo82 Last time I merged your work I used the magic "squash and merge" github button and it didnt work the way I though it would. So for this PR I would like it if you can squash your commits to one single commit.

File: src/providers/maven/mavenAPI.js (hunk 1 above)
Comment (fcrespo82) [reply src/providers/maven/mavenAPI.js]: That is true. I changed to getting the xml from a regex, this should make it platform agnostic

**Comment by fcrespo82** [comment]
Hey @pflannery I did what you asked...

Please check it to see if everything is ok.

**Comment by fcrespo82** [comment]
Can we follow up on this? Do you need anything change from my end to be able to merge?
I noticed that you open an issue at https://github.com/Microsoft/vscode-maven/issues/133 to validate this.
Should it be bumped to help validate the improved version?
Thanks again for letting me contribute.

**Pull request merged**